team/levels.tex

```diff
@@ -0,0 +1,105 @@
+  \hline
+    1 &
+    Demonstrates good understanding of basic data structures - hash tables, linked lists, trees. Can reason about algorithm complexity.
+    Applies relevant data structures in day to day activity, can implement a production quality. It might be not the most efficient or secure, but correct. &
+    Creates a design document based on the well defined scoped requirements and implements it. &
+    Applies basic security principles to program design. For example, can set up HTTPS and password based auth. &
+    Reports progress on a regular basis as required by team's operational requirements. Actively solicits feedback.&
+    Understands and reasons about networking concepts. Understands and can write production quality web servers. Understands common networking issues and troubleshooting techniques. &
+    Understands the usage of Posix and other APIs for Linux systems. Understands synchronisation primitives and their including reasoning about deadlocks, data races.
+    Can write basic systems level code understanding types of memory, allocation.
+    Understands inter process communication and can build systems leveraging it.
+    Can implement data race and deadlock free code using basic production guidelines - using synchronization primitives and properly sharing state between components of the system.&
+    Understands usage of compilers, interpreters, build tools used at the organisation. \\
+  \hline
+    2 & & &
+    Applies best practices industry security guidelines, like setting up strong TLS, can pick strong authentication and authorization mechanisms.&
+    &
+    Has more granular understanding of the networking design, for example can reason about using GRPC vs HTTPS-JSON and their networking and scalability trade-offs.
+    Can do the same for UDP vs TCP and lower level protocols.&
+    Can reason about performance implications and risks of using synchronisation primitives, understands granularity of locking, can debug and troubleshoot memory and synchronisation issues.&
+    Understands tools and compilers as advanced user - for example, can refactor Makefile to make it more efficient and parallel or select most optimal compilation flags of a complier for Linux and MacOS. \\
+  \hline
+    3 &
+    Uses advanced data structures and algorithms, such as consensus, gossip protocols to implement key product features.&
+    Collaborates with the team to scope requirements, based on good understanding of existing longer term product vision and estimates of the system design of a feature of a product.&
+    Understands security aspects of protocols on a deeper level, for example can explain differences between TLS 1.2 and 1.3 and security implications.
+    Understands common attack vectors for server side or client side applications and can build secure systems that will pass quality security audit that will uncover no critical system design errors.&
```

Comment:
```suggestion
    Understands common attack vectors for server side or client side applications and can build secure systems with no critical system design errors.&
```
I'd drop the part about security audit since it is implied by being secure.